modules/apps/transfer/internal/events.go

```diff
@@ -0,0 +1,115 @@
+	}
+
+	ctx.EventManager().EmitEvents(sdk.Events{
+		sdk.NewEvent(
+			types.EventTypePacket,
+			eventAttributes...,
+		),
+		sdk.NewEvent(
+			sdk.EventTypeMessage,
+			sdk.NewAttribute(sdk.AttributeKeyModule, types.ModuleName),
+		),
+	})
+}
+
+// EmitOnAcknowledgementPacketEvent emits a fungible token packet event in the OnAcknowledgementPacket callback
+func EmitOnAcknowledgementPacketEvent(ctx sdk.Context, packetData types.FungibleTokenPacketDataV2, ack channeltypes.Acknowledgement) {
+	ctx.EventManager().EmitEvents(sdk.Events{
+		sdk.NewEvent(
+			types.EventTypePacket,
+			sdk.NewAttribute(sdk.AttributeKeySender, packetData.Sender),
+			sdk.NewAttribute(types.AttributeKeyReceiver, packetData.Receiver),
+			sdk.NewAttribute(types.AttributeKeyTokens, types.Tokens(packetData.Tokens).String()),
+			sdk.NewAttribute(types.AttributeKeyMemo, packetData.Memo),
+			sdk.NewAttribute(types.AttributeKeyAck, ack.String()),
+		),
+		sdk.NewEvent(
+			sdk.EventTypeMessage,
+			sdk.NewAttribute(sdk.AttributeKeyModule, types.ModuleName),
+		),
+	})
+
+	switch resp := ack.Response.(type) {
+	case *channeltypes.Acknowledgement_Result:
+		ctx.EventManager().EmitEvent(
+			sdk.NewEvent(
+				types.EventTypePacket,
+				sdk.NewAttribute(types.AttributeKeyAckSuccess, string(resp.Result)),
+			),
+		)
+	case *channeltypes.Acknowledgement_Error:
+		ctx.EventManager().EmitEvent(
+			sdk.NewEvent(
+				types.EventTypePacket,
+				sdk.NewAttribute(types.AttributeKeyAckError, resp.Error),
+			),
+		)
+	}
+}
+
+// EmitOnTimeoutPacketEvent emits a fungible token packet event in the OnTimeoutPacket callback
+func EmitOnTimeoutEvent(ctx sdk.Context, packetData types.FungibleTokenPacketDataV2) {
+	ctx.EventManager().EmitEvents(sdk.Events{
+		sdk.NewEvent(
+			types.EventTypeTimeout,
+			sdk.NewAttribute(types.AttributeKeyReceiver, packetData.Sender),
+			sdk.NewAttribute(types.AttributeKeyRefundTokens, types.Tokens(packetData.Tokens).String()),
+			sdk.NewAttribute(types.AttributeKeyMemo, packetData.Memo),
+		),
+		sdk.NewEvent(
+			sdk.EventTypeMessage,
+			sdk.NewAttribute(sdk.AttributeKeyModule, types.ModuleName),
+		),
+	})
+}
+
+// EmitTransferEvent emits a ibc transfer event on successful transfers.
+func EmitTransferEvent(ctx sdk.Context, sender, receiver string, tokens types.Tokens, memo string) {
+	ctx.EventManager().EmitEvents(sdk.Events{
+		sdk.NewEvent(
+			types.EventTypeTransfer,
+			sdk.NewAttribute(types.AttributeKeySender, sender),
+			sdk.NewAttribute(types.AttributeKeyReceiver, receiver),
+			sdk.NewAttribute(types.AttributeKeyTokens, tokens.String()),
+			sdk.NewAttribute(types.AttributeKeyMemo, memo),
+		),
+		sdk.NewEvent(
+			sdk.EventTypeMessage,
+			sdk.NewAttribute(sdk.AttributeKeyModule, types.ModuleName),
+		),
+	})
+}
+
+// EmitDenomTraceEvent emits a denomination trace event in the OnRecv callback.
+func EmitDenomTraceEvent(ctx sdk.Context, traceHash string, voucherDenom string) {
```

Comment:
We could already call this `EmitDenomEvent` ahead of the renaming coming [here](https://github.com/cosmos/ibc-go/pull/6466). I can also do it in my PR, if your is merged first.